docs/rules/no-function-iterator.md

```diff
@@ -0,0 +1,92 @@
+# Prevents passing a function directly to iterator methods
+
+Prevents passing a function directly to an iterator method to make it more clear with the function accepts.
+
```

Comment:
The intro should include why it's a useful rule. See: https://github.com/sindresorhus/eslint-plugin-unicorn/issues/91#issue-219641506 I don't think most realize the issue with doing this.